File: backend/app/schemas/refi.py
import uuid
from datetime import datetime
from typing import Literal

from pydantic import BaseModel


class RefiPackageResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: uuid.UUID
    deal_id: uuid.UUID
    status: str
    amount_cents: int | None
    duration_months: int | None
    monthly_payment_cents: int | None
    risk_score: float | None
    risk_band: str | None
    sent_at: datetime | None
    created_at: datetime
    updated_at: datetime
    company_name: str | None = None
    deal_public_id: str | None = None


class FinancierDecisionRequest(BaseModel):
    decision: Literal["approved", "rejected"]
    reason: str | None = None


class FinancierDecisionResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: uuid.UUID
    refi_package_id: uuid.UUID
    decision: str
    reason: str | None
    decided_at: datetime

File: web/lib/types/refi.ts
export interface RefiPackage {
  id: string
  deal_id: string
  status: 'draft' | 'sent' | 'financier_approved' | 'financier_rejected'
  amount_cents: number | null
  duration_months: number | null
  monthly_payment_cents: number | null
  risk_score: number | null
  risk_band: string | null
  sent_at: string | null
  created_at: string
  updated_at: string
}

export interface FinancierDecisionRequest {
  decision: 'approved' | 'rejected'
  reason?: string
}

export interface FinancierDecision {
  id: string
  refi_package_id: string
  decision: 'approved' | 'rejected'
  reason: string | null
  decided_at: string
}
